jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/RequestLogTest.java

```diff
@@ -359,40 +349,118 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
+
+                // We should see a requestlog entry for the original 202 response
+                String reqlog = requestLogLines.poll(3, TimeUnit.SECONDS);
+                assertThat("RequestLog", reqlog, containsString("method:GET|uri:%s|header[x-name]:actual|status:202".formatted(expectedURI)));
+            }
+        }
+        finally
+        {
+            LifeCycle.stop(server);
+        }
+    }
 
-                ByteArrayOutputStream outBuf = new ByteArrayOutputStream();
-                IO.copy(in, outBuf);
-                String response = outBuf.toString(UTF_8);
-                if (LOG.isDebugEnabled())
-                    LOG.debug("--Response--\n" + response);
+    /**
+     * Test where the request local-address and remote-address is accessible during RequestLog.
+     * Requires that the EndPoint is closed before the RequestLog can execute.
+     */
+    @Test
+    public void testLogRemoteAndLocalAddressesAfterClose() throws Exception
+    {
+        Server server = null;
+        try
+        {
+            BlockingArrayQueue<String> requestLogLines = new BlockingArrayQueue<>();
+
+            RequestLog requestLog = (request, response) ->
+            {
+                SocketAddress remoteAddress = request.getConnectionMetaData().getRemoteSocketAddress();
+                SocketAddress localAddress = request.getConnectionMetaData().getLocalSocketAddress();
+                requestLogLines.add(String.format("method:%s|uri:%s|remote-addr:%s|local-addr:%s|status:%d",
+                    request.getMethod(), request.getHttpURI(), remoteAddress, localAddress, response.getStatus()));
+            };
+
+            Handler handler = new Handler.Abstract()
+            {
+                @Override
+                public boolean handle(Request request, Response response, Callback callback) throws Exception
+                {
+                    response.setStatus(202);
+                    response.getHeaders().put("X-RemoteAddr", Objects.toString(request.getConnectionMetaData().getRemoteSocketAddress(), "<null>"));
+                    response.getHeaders().put("X-LocalAddr", Objects.toString(request.getConnectionMetaData().getLocalSocketAddress(), "<null>"));
+                    response.getHeaders().put(HttpHeader.CONTENT_TYPE, "text/plain; charset=utf-8");
+
+                    String msg = "Got %s to %s%n".formatted(request.getMethod(), request.getHttpURI());
+                    Callback testCallback = Callback.from(callback, () ->
+                    {
+                        EndPoint endPoint = request.getConnectionMetaData().getConnection().getEndPoint();
+                        // Close connection
+                        endPoint.close();
+                        // Wait for endpoint to be closed
+                        Awaitility.await().atMost(Duration.ofSeconds(5)).until(() -> !endPoint.isOpen());
```

Comment:
Please `static import Awaitility.await` like we do with other testing facilities.